staging/src/k8s.io/api/batch/v1/types.go

```diff
@@ -46,10 +46,11 @@ const (
 	// The Job labels will use batch.kubernetes.io as a prefix for all labels
 	// Historically the job controller uses unprefixed labels for job-name and controller-uid and
 	// Kubernetes continutes to recognize those unprefixed labels for consistency.
-	JobNameLabel = labelPrefix + "job-name"
+	JobNameLabel             = labelPrefix + "job-name"
+	LegacyControllerUidLabel = "controller-uid"
 	// ControllerUid is used to programatically get pods corresponding to a Job.
 	// There is a corresponding label without the batch.kubernetes.io that we support for legacy reasons.
-	ControllerUidLabel = labelPrefix + "controller-uid"
+	ControllerUidLabel = labelPrefix + LegacyControllerUidLabel
```

Comment:
TBH We really want one to use the `ControllerUidLabel` and not the legacy name.  Could you just use ControllerUidLabel?

Reply:
Sorry i have not get it.do you mean the `name` or this change.

I assume you are talking about the change of this constant. If kubernetes does not accept this constant, then there may be a "controller-uid" constant in each project. Obviously, the effect of this constant existing in the kubernetes project is the most ideal.

Reply:
I do not understand what you mean "by accept this constant".  Generally, k8s is trying to phase out naked labels for k8s scoped ones.

There is a deprecation of the label "controller-uid" so referencing this label is discouraged.  Ideally, one should use  ControllerUidLabel.  

We already have a deprecation notice for `job-name` and `controller-uid`.

https://kubernetes.io/docs/reference/labels-annotations-taints/#controller-uid

Reply:
We do not want projects to continue to depend on the `controller-uid` label.
You should use `batch.kubernetes.io/controller-uid`.
This is final.
/close